GeneralisedFilters/src/algorithms/particles.jl

```diff
@@ -0,0 +1,232 @@
+export BootstrapFilter, BF
+export ParticleFilter, PF, AbstractProposal
+
+# import SSMProblems: distribution, simulate, logdensity
+
+abstract type AbstractProposal end
+
+function distribution(
```

Comment:
Are these meant to be `SSMProblems.distribution`? Currently they are different functions that belong to `GeneralisedFilters`. Could be confusing/lead to conflicts.